pkg/subctl/cmd/validate_fw_metrics.go

```diff
@@ -0,0 +1,103 @@
+)
+
+var validateFirewallMetricsCmd = &cobra.Command{
+	Use:   "metrics",
+	Short: "Validate if firewall allows metrics on the Gateway nodes.",
+	Long:  "This command checks whether firewall configuration allows metrics to be read from the Gateway nodes.",
+	Run:   validateFirewallMetricsConfig,
+}
+
+func init() {
+	addValidateFWConfigFlags(validateFirewallMetricsCmd)
+	validateFirewallConfigCmd.AddCommand(validateFirewallMetricsCmd)
+}
+
+func validateFirewallMetricsConfig(cmd *cobra.Command, args []string) {
+	configs, err := getMultipleRestConfigs(kubeConfig, kubeContext)
+	exitOnError("Error getting REST config for cluster", err)
+
+	for _, item := range configs {
+		status.Start(fmt.Sprintf("Validating the firewall configuration to check if metrics port (8080)"+
+			" is allowed in cluster %q", item.clusterName))
+		validateFirewallMetricsConfigWithinCluster(item)
+		status.End(status.ResultFromMessages())
+	}
+}
+
+func validateFirewallMetricsConfigWithinCluster(item restConfig) {
+	clientSet, err := kubernetes.NewForConfig(item.config)
+	if err != nil {
+		message := fmt.Sprintf("Error creating API server client: %s", err)
+		status.QueueFailureMessage(message)
+		return
+	}
+
+	podCommand := fmt.Sprintf("timeout %d %s", validationTimeout, TCPSniffMetricsCommand)
+	sPod, err := spawnSnifferPodOnGatewayNode(clientSet, namespace, podCommand)
+	if err != nil {
+		message := fmt.Sprintf("Error while spawning the sniffer pod on the GatewayNode: %v", err)
+		status.QueueFailureMessage(message)
+		return
+	}
+
+	defer sPod.DeletePod()
+	gatewayPodIP := sPod.Pod.Status.HostIP
+	podCommand = fmt.Sprintf("for i in $(seq 10); do timeout 2 nc -p 9898 %s 8080; done", gatewayPodIP)
+	cPod, err := spawnClientPodOnNonGatewayNode(clientSet, namespace, podCommand)
+	if err != nil {
+		message := fmt.Sprintf("Error while spawning the client pod on non-Gateway node: %v", err)
+		status.QueueFailureMessage(message)
+		return
+	}
+
+	defer cPod.DeletePod()
+	if err = cPod.AwaitPodCompletion(); err != nil {
+		message := fmt.Sprintf("Error while waiting for client pod to be finish its execution: %v", err)
+		status.QueueFailureMessage(message)
+		return
+	}
+
+	if err = sPod.AwaitPodCompletion(); err != nil {
+		message := fmt.Sprintf("Error while waiting for sniffer pod to be finish its execution: %v", err)
+		status.QueueFailureMessage(message)
+		return
+	}
+
+	// Verify that tcpdump output (i.e, from snifferPod) contains the HostIP of clientPod
+	if !strings.Contains(sPod.PodOutput, cPod.Pod.Status.HostIP) {
+		message := fmt.Sprintf("The tcpdump output from the sniffer pod does not contain the"+
+			" client pod HostIP. Please check that your firewall configuration allows TCP/8080 traffic"+
+			" on the %q node.", sPod.Pod.Spec.NodeName)
+		status.QueueFailureMessage(message)
+		return
+	}
+
+	status.QueueSuccessMessage("The firewall configuration for prometheus metrics is working fine.")
```

Comment:
```suggestion
	status.QueueSuccessMessage("Prometheus metrics can be retrieved from gateway nodes.")
```

I’m a bit late to the party on this series, but we’re not really checking the firewall — we don’t know that there is one — we’re checking connectivity.

Reply:
That's true @skitt. We are checking if the metrics port (8080) on the Gateway node is accessible. This is an indirect way of ensuring that even if the firewall is enabled, its properly configured to allow the port. I'll update the comment. Thanks for reviewing.

Reply:
Does this mean that the overall command should be less about the (potential) firewall?